Milestone 4

okikiola11 opened the pull request:
- Created a new logic folder inside src and initialized (calculate.js and operate.js)

- Implemented the calculate.js module
-- it contains the calculate method
-- the calculate method accepts a calculator data object and a button name as arguments.
-- the calculator data object contains total, next and operation as properties
-- buttonName argument mutates the calculator’s data object

- Implemented the operate.js module
-- it implements the Operate method.
-- Operate function accepts numberOne, numberTwo and operation as arguments.
-- it performs a simple math operation.

- Added calculate.js to the App component
-- Imported the calculate function.

okikiola11 committed:
Add the logic folder- which is the heart of the calculator

okikiola11 committed:
Implement calculate.js
calculate method should accept a calculator data object and a button name as arguments
calculator data object should contain three properties: total, next and operation
Implement the operate.js module
The operate function accept numberOne, numberTwo and operation as arguments

okikiola11 committed:
Remove commented code from App.js

okikiola11 committed:
Remove the div element in the App.js file
Use only React.Fragment to wrap components

trekab commented on src/components/App.js:
```diff
@@ -1,13 +1,13 @@
-  </div>
+  <>
+    <DisplayCalculator />
+    <ButtonPanel />
+    <Calculate />
```
It appears that you used `Calculate` as a react component. It should be used as a function that other components will rely on to perform calculations. It is therefore better to remove the `<Calculate/>` line.
```suggestion
    <ButtonPanel />
```

trekab commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,29 @@
+import Operate from './operate';
+
+const Calculate = (dataObj, buttonName) => {
+  let { total, next, operation } = dataObj;
+  const operators = ['+', '÷', 'x', '-', '%'];
+
```
It is better to name your javascript functions starting with a lower case letter. Check out some highlighted JS coding conventions [here](https://www.w3schools.com/js/js_conventions.asp). __Note:__ This applies to the `operate` function as well.

trekab commented on src/logic/operate.js:
```diff
@@ -0,0 +1,26 @@
+  if (operation === '÷') {
+    result = num1.div(num2).toString();
+  }
+  if (operation === '%') {
+    result = num1.mod(num2).toString();
+  }
```
The `%` modifier should  turn the number into a percentage based number, `e.g. 95 % == .95` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png))

trekab reviewed (changes requested):
#### Changes Required! :recycle: 
Well done @okikiola11! So far so good. There are a few changes for you to work on to complete this milestone. These include:

- [x] As indicated in the code comments below, try to name the operate and calculate functions more appropriately to follow industry standards.
- [x] Fix the issue where you used `Calculate` as a component in the `App` component. (Details highlighted in the code comment below)
- [x] Fix the logic for the `%` operator as illustrated below.

__Please make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__[Treasure Kabareebe](https://trekab.github.io)__

okikiola11 committed:
Rename functions to use JS convention
Remove the Calculate method from the App.js file
The calculate method is used as a function

okikiola11 committed:
Fix eslint errors

okikiola11 committed:
Fix the modulus error

bellom commented on src/components/App.js:
```diff
@@ -1,12 +1,10 @@
 import DisplayCalculator from './DisplayCalculator';
 import ButtonPanel from './ButtonPanel';
-import Calculate from '../logic/calculate';
```
Oops, you're not supposed to remove this line, you will need it in the next milestone, kindly add it back before merging, I trust you can do this. 👀  on you!....

bellom reviewed (approved):
## Hi @okikiola11  :clap: 

**Great work making all the necessary corrections from the previous review.**

**Do well to check my comment and make the little changes before merging.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

bellom commented on src/components/App.js:
```diff
@@ -1,5 +1,6 @@
 import DisplayCalculator from './DisplayCalculator';
 import ButtonPanel from './ButtonPanel';
+import calculate from '../logic/calculate';
```
👍

okikiola11 committed:
Add the calculate function to the App.js
Fix TSE review

okikiola11 committed:
Remove the calculate function to the App.js

okikiola11 committed:
Remove the calculate function to the App.js
